### DISCUSSION TITLE
Minio presigned url upload issue

### DISCUSSION THREAD

**Discussion by mehditounsi** [minio/minio#13926] [Q&A] [answered]
Hi Minio experts , I have an issue with the Minio presigned url , I've been able to get the url and to use the PUT method to insert my file into my Minio bucket but i could not open it especially when it is a jpg , a png or a pdf file because it's automatically modified by Minio who adds a header and a footer to the file what makes it unreadable as an image

exemple of header :
----------------------------591397828093304071314847
Content-Disposition: form-data; name="file"; filename="y.png"
Content-Type: image/png

here's what i got when i download it using the presignedGetObject
the problem is with the metadata added by minio
![Capture](https://user-images.githubusercontent.com/76108460/146332432-6a31e9c1-3981-4769-88c5-4dd419dd831f.PNG)

I'm using Nodejs , here is my code : 

service.js: 

exports.presignedurl = async (filename) =>{
  return await minioClient.presignedPutObject(process.env.MINIO_BUCKET,filename)
}
exports.getpresignedurl = async (filename) =>{
  return await minioClient.presignedGetObject(process.env.MINIO_BUCKET,filename, 24 * 60 * 60)
}

controller.js : 

exports.presignedurl = async (req = null,res = null) =>{
try {
  let filename = req.query.filename
  let result = await StorageService.presignedurl(filename)
  res.status(200).json(result)
} catch (err) {
  res.status(405).send({ error: err.message, code: err.code });
}
}
exports.getpresignedurl = async (req = null,res = null) =>{
  try {
    let filename = req.query.filename
    let result = await StorageService.getpresignedurl(filename)
    res.status(200).json(result)
  } catch (err) {
    res.status(405).send({ error: err.message, code: err.code });
  }
  }

**Comment by prakashsvmx** [comment, marked as answer]
`presignedGetObject`  returns a stream. So check the code accordingly.

so  the content needs to be accessed accordingly: Please have a look at the functional test.

```


 client.presignedGetObject(bucketName, _1byteObjectName, 1000, (e, presignedUrl) => {
        if (e) return done(e)
        var transport = http
        var options = _.pick(url.parse(presignedUrl), ['hostname', 'port', 'path', 'protocol'])
        options.method = 'GET'
        if (options.protocol === 'https:') transport = https
        var request = transport.request(options, (response) => {
          if (response.statusCode !== 200) return done(new Error(`error on put : ${response.statusCode}`))
          var error = null
          response.on('error', e => done(e))
          response.on('end', () => done(error))
          response.on('data', (data) => {
            if (data.toString() !== _1byte.toString()) {
              error = new Error('content mismatch')
            }
          })
        })
        request.on('error', e => done(e))
        request.end()
      })
```

**Reply by mehditounsi** [reply]
thank you for your response but it didn't work for me , I still have the metadata added everytime i use the presignedPutObject in my files 
it can't be a version problem ?
